tests/integration/test_overrides.yaml

```
@@ -0,0 +1,6 @@
+trainer#max_steps: 5
+trainer#num_sanity_val_steps: 0
+trainer#accelerator: cpu
+system#batch_size: 16
+system#num_workers: 2
+trainer#callbacks: null
```

To reduce testing time, we should simplify the tests and only focus on testing if the code runs. 

Additionally, we could implement slower tests that run for multiple epochs with a fixed seed. These tests will ensure that the metrics remain consistent, indicating that the code is functioning correctly and the network trains correctly. But such a test will take several minutes and is annoying if you want to just add a simple PR. 

```suggestion
trainer#max_steps: 2
trainer#num_sanity_val_steps: 0
trainer#accelerator: cpu
system#batch_size: 4
system#num_workers: 2
trainer#callbacks: null
```

---

I agree! We should have two sets of tests. One that runs by default on PR and then another that we can trigger manually when we want to merge more breaking changes (this will be the longer multiple epoch ones). 

For the latter, `pytest` provides some nice functionality to tag tests with different markers. If you see the integration test here: https://github.com/project-lighter/lighter/blob/5d02775987b74a88076f9dba14bfd366f2d2c56e/tests/integration/test_cifar.py#L13

I've marked it as slow. We can set pytest markers for each of the two sets accordingly and run them as we need